qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -283,9 +294,15 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return dag
 
 
-def _build_gate_lengths(props):
+def _build_gate_lengths(props, target):
```

Comment:
Is there a reason this needs to take two arguments? It currently doesn't seem to be the case that information from `props` and `target` is ever merged. The method seems to focus on one source of information and just handles each differently. If speed of e.g. `isinstance` is a concern, perhaps these could be keyword arguments instead?

Reply:
That's a good call, I was too concentrated on keeping everything separated between target and props through everything. But a single isinstance call here is fine, it's only called once per run.

Reply:
Actually looking at this a bit more, the way it gets called is a bit inconvenient to have it pass either a properties or target because we'd need an if block on the caller side like:

```python
if self.target is not None:
    _build_gate_lengths(self._target)
else:
    _build_gate_lengths(self._backend_props)
```

as opposed to what is there right now:

```python
_build_gate_lengths(self._backend_props, self._target)
```

So I think going with a kwarg defaulting to None for both arguments here makes more sense.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7775/commits/985ddce0bd9de0374a8b6e9ff5b06539da1efc0f